app/controllers/map_proxy_controller.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require "faraday"
+
+class MapProxyController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
+  def proxy
+    faraday_response = proxy_to_ordnance_survey(build_os_url)
+    configure_response_headers
+    send_response(faraday_response)
+  end
+
+  private
+
+  def build_os_url
+    os_path = request.fullpath.sub("/map_proxy", "")
+    "https://api.os.uk#{os_path}"
+  end
```

Comment:
Maybe `build_os_url` could just be `os_url` since it returns the value.

```suggestion
  def proxy
    faraday_response = proxy_to_ordnance_survey(os_url)
    configure_response_headers
    send_response(faraday_response)
  end

  private

  def os_url
    os_path = request.fullpath.sub("/map_proxy", "")
    "https://api.os.uk#{os_path}"
  end
```